README.md

```diff
@@ -310,6 +322,59 @@ This selector is applicable when "deviceType" is "auxNetDevice".
 
 [//]: # (The tables above generated using: https://ozh.github.io/ascii-tables/)
 
+#### AdditionalInfo field
+
+This field defines a method to add information as part of the environment variable the sriov-network-device-plugin injects to the container.
+
+Examples:
+```json
+"additionalInfo": {
+   "*": {
```

Comment:
lets discuss this bit again.

additionalInfo allows to add additional information for device upon allocation. this information is exposed via the new _INFO env variable.

the API allows to provide information (in the form of key value string pairs) either globally to all devices or per device.

I have some doubts about the usability of the latter form as when a user creates a resource pool, it has no knowledge on the specific device addresses that will be part of the pool, it uses selectors to choose them.

the "per-device" bit here contradicts that. to inject information on a per device basis would mean the user needs to know the device that will be chosen. (moreover they need to be the same for every device plugin which will use the configuration)
 
so i wonder how usable is this (per-device) case in general.

extra info per device can be implemented in a bit of a different manner.
1. have folder on the system that will contain per device information (root owned) in the form of files
2. these files are written by some external entity
3. folder mounted to device plugin
4. info provider reads this folder, if it has a file for the device it will read it and provide the information

just throwing some ideas as im abit concerned about the usability of it.

Reply:
the per device is more for a controller sitting on top of the device plugin. I am not expecting someone directly using the per-device configuration

for example in the case of a specific value for every device the operator will create the virtual functions for a specific device for example an accelerator card so he knows the PCI address for every virtual function and then he will create the additionalInfo with specific device information for every PCI address.
this is also per node configuration as we can tell every device plugin to use its own configuration

Reply:
OK,

As i dont have a better idea on how to enable per-device configuration AND after thinking on it, i dont think we want another mechanism to inject additional information (via files). It makes sense that if per-device configuration are needed then the ID of these devices need to be known by someone (human or k8 operator).

this can then be expressed via a per node  sriov dp config.